### PR TITLE
Add CircleCI configuration

### PR DESCRIPTION
**Pull request by andrecardoso**

### CODE REVIEW

**Comment by andrecardoso** [comment]
[(2) Set up build and deployment (staging and production) CI](https://trello.com/c/SLmIYL9n/19-2-set-up-build-and-deployment-staging-and-production-ci)

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,178 @@
+version: 2.1
+
+jobs:
+  preconditions:
+    docker: &BUILDIMAGE
+      - image: jenkinsrise/cci-v2-components:0.0.4
+    steps:
+      - run: |
+          if [ -z "$COMPONENT_NAME" ]
```
Comment (olegrise) [inline .circleci/config.yml]: @andrecardoso Do you know what it's done that way? Why COMPONENT_NAME has to be configure in CircleCI and not included into `config.yml` itself?

Comment (andrecardoso) [reply .circleci/config.yml]: I don't know. I followed [rise-data-financial](https://github.com/Rise-Vision/rise-data-financial/blame/master/.circleci/config.yml#L9). @stulees any reason we are doing this way?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+  build:
+    docker: *BUILDIMAGE
+    steps:
+      - checkout
+      - attach_workspace:
+          at: .
+      - restore_cache:
+          key: node-cache-{{ checksum "package.json" }}
+      - run: npm run build
```
Comment (olegrise) [inline .circleci/config.yml]: @andrecardoso I see that financial component [updates the version file](https://github.com/Rise-Vision/rise-data-financial/blob/master/.circleci/config.yml#L43-L47) and we don't do that. Don't we need it too?

Comment (andrecardoso) [reply .circleci/config.yml]: They are using the version file only for [logging purposes](https://github.com/Rise-Vision/rise-data-financial/blob/master/src/rise-data-financial.js#L153). As we are not doing any logging so far, I decided to not include it. We can add the version file when we add logs.

File: .circleci/config.yml (hunk 1 above)
Comment (andrecardoso) [reply .circleci/config.yml]: Per Santiago, this was inherited from widgets configuration. I'll simplify it and declare the component name directly in the configuration. Thanks!

Comment (stulees) [reply .circleci/config.yml]: @andrecardoso @olegrise It's not imperative that `COMPONENT_NAME` be an environment variable in CCI, I agree with simplifying it, cheers.

**Pull request merged**